### PR TITLE
ongoing vampire ability balance/rework attempt

### PR DESCRIPTION
**Pull request by jack-avery**
## About The Pull Request

title

## Why It's Good For The Game

vampires are woefully underpowered or overpowered in separate areas, me and artur are working out better abilities with feedback. please give feedback

## Changelog
:cl:
fix: cloak sometimes not restoring initial move intent
tweak: mesmerize (line of sight checking system and remove progress bar)
balance: nerf: lunge, mesmerize
/:cl:

### CODE REVIEW

**Comment by jack-avery** [comment]
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11268/commits/39e25935c742f268eb53ac6467a6b9c64b4c82c0 makes me very angry and i wish it wasn't a part of this pr

**Comment by Linzolle** [comment]
not only is the way you made mesmerise not have a bar very dumb and hacky, it doesn't even work correctly the way you described it. it'll still break after power_time deciseconds. (for var/i in 1 to 6) will stop looping after 6 times.

**Comment by jack-avery** [comment]
> not only is the way you made mesmerise not have a bar very dumb and hacky

i tried making a proc to callback to and break the do_mob but it didn't seem to work no matter how i put it. if you want to help me with that, i'm all ears.


> it doesn't even work correctly the way you described it.

found the issue. since there's not really any sort of debug window i can print values to that i know of right now i just rearranged the code and it seems to work fine now.


> (for var/i in 1 to 6) will stop looping after 6 times.

that's the point. each loop is 5 deciseconds long resulting in a 2.5 second windup (oops, logic error, gonna fix that next) before the full hardstun comes into effect (if(i==6)
i guess your issue is with how i arranged it, and could have just put everything after the loops. this works fine. in my opinion. even though that's a pretty piss poor excuse.

**Comment by jack-avery** [comment]
thanks kev for informing me of the signal proc now it actually functions like a normal ability

File: code/modules/antagonists/bloodsucker/powers/lunge.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	button_icon_state = "power_lunge"
 	bloodcost = 10
 	cooldown = 120
-	target_range = 5
+	target_range = 3 // artur was right
```
Comment (Arturlang) [inline code/modules/antagonists/bloodsucker/powers/lunge.dm]: The comment isnt needed. Dont immortalize your own mistake.

File: code/modules/antagonists/bloodsucker/powers/mesmerize.dm [hunk 2]
```diff
@@ -73,7 +73,7 @@
 			to_chat(owner, "<span class='warning'>You're too far outside your victim's view.</span>")
 		return FALSE
 
-	if(target.has_status_effect(STATUS_EFFECT_MESMERIZE)) // ?
+	if(target.has_status_effect(STATUS_EFFECT_MESMERIZE)) // ignores facing once the windup has started (now i know why i put this here!)
```
Comment (Arturlang) [inline code/modules/antagonists/bloodsucker/powers/mesmerize.dm]:
```suggestion
	if(target.has_status_effect(STATUS_EFFECT_MESMERIZE)) // ignores facing once the windup has started

File: code/modules/antagonists/bloodsucker/powers/mesmerize.dm [hunk 3]
```diff
@@ -88,39 +88,53 @@
...
+		to_chat(user, "<span class='warning'>[target] has escaped your gaze!</span>")
+		UnregisterSignal(target, COMSIG_MOVABLE_MOVED)
+
 /datum/action/bloodsucker/targeted/mesmerize/FireTargetedPower(atom/A)
 	// set waitfor = FALSE   <---- DONT DO THIS!We WANT this power to hold up ClickWithPower(), so that we can unlock the power when it's done.
 	var/mob/living/carbon/target = A
 	var/mob/living/user = owner
 
 	if(istype(target))
+		success=TRUE
```
Comment (Arturlang) [inline code/modules/antagonists/bloodsucker/powers/mesmerize.dm]:
```suggestion
		success = TRUE
```

**Review by Ghommie — APPROVED** [review]

**Pull request merged**